solutions/task_3.py

```diff
@@ -6,57 +6,63 @@
 from datetime import datetime, timedelta
 
 
-def add_2(func):
-    def wrapper(a, b):
-        return func(a, b) + 2
-    return wrapper
+CACHE = {}
+CACHE_LIFE_TIME = 1
 
 
-def add_n(n):
-    def wrapper_1(func):
-        def wrapper_2(a, b):
-            return func(a, b) + n
-        return wrapper_2
-    return wrapper_1
+def add_2(func):
+    def wrapped(*args, **kwargs):
+        return func(*args, **kwargs) + 2
+    return wrapped
 
 
-CACHE = {}
+def add_n(n):
+    def wrapper(func):
+        def wrapped(*args, **kwargs):
+            return func(*args, **kwargs) + n
+        return wrapped
+    return wrapper
 
 
 def cached(cache_life_time):
     def wrapper_1(func):
         CACHE[func] = {}
 
         def wrapper_2(*args, **kwargs):
-            if str(args) + str(kwargs) not in CACHE[func].keys():
-                CACHE[func][str(args) + str(kwargs)] = [None, None]
-                CACHE[func][str(args) + str(kwargs)][0] = func(*args, **kwargs)
-                CACHE[func][str(args) + str(kwargs)][1] = datetime.now()
-                return CACHE[func][str(args) + str(kwargs)][0]
-            if (datetime.now() - CACHE[func][str(args)+str(kwargs)][1]).total_seconds() < cache_life_time:
-                return CACHE[func][str(args)+str(kwargs)][0]
-            CACHE[func][str(args)+str(kwargs)][0] = func(*args, **kwargs)
-            CACHE[func][str(args)+str(kwargs)][1] = datetime.now()
-            return CACHE[func][str(args)+str(kwargs)][0]
+            key = str(args) + str(kwargs)
+            if key not in CACHE[func]:
+                CACHE[func][key] = [func(*args, **kwargs), time.time()]
+                return CACHE[func][key][0]
+            if (time.time() - CACHE[func][key][1]) < cache_life_time:
+                return CACHE[func][key][0]
+            else:
+                CACHE[func][key][0] = func(*args, **kwargs)
+                CACHE[func][key][1] = time.time()
+                return CACHE[func][key][0]
```

Comment:
If you will improve this code (lines 39-41) you will get the same code as above:
```
CACHE[func][key] = [func(*args, **kwargs), time.time()]
return CACHE[func][key][0]
```
Would you like to improve this moment? (I suppose yes)